File: backend/src/main/java/dev/datn/foco/service/PaymentService.java
package dev.datn.foco.service;

import dev.datn.foco.dto.request.PaymentRequest;
import dev.datn.foco.dto.respone.PaymentResponse;

import java.util.List;

public interface PaymentService {

    PaymentResponse createPayment(PaymentRequest request);

    PaymentResponse getPaymentById(Long id);

    List<PaymentResponse> getPaymentsByOrder(Long orderId);

    PaymentResponse getPaymentByTransactionId(String transactionId);

    PaymentResponse updatePaymentStatus(Long id, String status);

    PaymentResponse confirmPayment(Long id);

    void deletePayment(Long id);
}

File: frontend/src/services/paymentService.js
import api from '../config/api';

const paymentService = {
  createPayment: async (paymentData) => {
    const response = await api.post('/payments', paymentData);
    return response.data;
  },

  getPaymentById: async (id) => {
    const response = await api.get(`/payments/${id}`);
    return response.data;
  },

  getPaymentsByOrder: async (orderId) => {
    const response = await api.get(`/payments/order/${orderId}`);
    return response.data;
  },

  confirmPayment: async (id) => {
    const response = await api.patch(`/payments/${id}/confirm`);
    return response.data;
  },

  updatePaymentStatus: async (id, status) => {
    const response = await api.patch(`/payments/${id}/status?status=${status}`);
    return response.data;
  },
};

export default paymentService;

